[DM-3311] LWM2M agent process only one of a multiple Resource Instances sent in a row (#1782)
SoftwareAG/c8y-docs: content/change-logs/ui-c8y-1018-503-55-sent-in-a-row-lwm2m-resource-instance-not-processed.md

---
date: ""

> title: LWM2M agent process only one of a multiple Resource Instances sent in a row

Reviewer: ```suggestion
title: LWM2M agent processes all multiple resource instance IDs correctly
```

product_area: Device management & connectivity
change_type:
  - value: change-2c7RdTdXo4

>     label: Improvement

Reviewer 2: ```suggestion
    label: Fix
```

component:
  - value: component-1KLUzmqfe
    label: LWM2M
build_artifact:
  - value: tc-pjJiURv9Y
    label: ui-c8y
ticket: DM-3311
version: 1018.540.66
---

> When a LWM2M device sends a composite request with multiple Resource Instance IDs from a same Resource ID and this Resource ID has an additional action defined, only one of these Resource Instances values was sent to this additional action for processing. The same behavior is observed when you try to do a composite read (`cread`) device operation with a row of Resource Instance IDs. These issues are resolved and all requested Resource Instance IDs are processed

Reviewer: ```suggestion
When a LWM2M device sends a composite request with multiple resource instance IDs from the same resource ID and this resource ID has an additional action defined, only one of these resource instance values was sent to this additional action for processing. The same behavior was observed when you try to do a composite read (`cread`) device operation with a row of resource instance IDs. These issues are resolved and all requested resource instance IDs are now processed.
```
